File: backend_deprecated/src/models/Survey.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const S = DataTypes.STRING;
const F = DataTypes.DOUBLE;

const Survey = sequelize.define('Survey', {
  id:   { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
  uuid: { type: S, allowNull: false, unique: true },

  // GP identity (names stored for export even if FK becomes stale)
  stateId:          { type: DataTypes.INTEGER, allowNull: true },
  stateName:        { type: S, allowNull: true },
  districtId:       { type: DataTypes.INTEGER, allowNull: true },
  districtName:     { type: S, allowNull: true },
  blockId:          { type: DataTypes.INTEGER, allowNull: true },
  blockName:        { type: S, allowNull: true },
  gramPanchayatId:  { type: DataTypes.INTEGER, allowNull: true },
  gramPanchayatName:{ type: S, allowNull: true },
  gramPanchayatCode:{ type: S, allowNull: true },

  phase:        { type: S, allowNull: true },
  surveyVendor: { type: S, allowNull: true },
  surveyDate:   { type: S, allowNull: true },
  surveyDone:   { type: S, defaultValue: 'YES' },
  remarks:      { type: DataTypes.TEXT, allowNull: true },

  // Section A — Original Location
  origLocationType: { type: S, allowNull: true },
  origInfraStatus:  { type: S, allowNull: true },
  origElectricity:  { type: S, allowNull: true },
  origPowerHours:   { type: S, allowNull: true },
  origSolar:        { type: S, allowNull: true },
  origEarthing:     { type: S, allowNull: true },
  origLat:          { type: F, allowNull: true },
  origLong:         { type: F, allowNull: true },

  // Section B — Current Location
  currentLocation: { type: S, allowNull: true },
  currentPermTemp: { type: S, allowNull: true },
  currentLat:      { type: F, allowNull: true },
  currentLong:     { type: F, allowNull: true },

  // Section C — GP Bhawan
  gpBhawanAvailable:   { type: S, allowNull: true },
  gpBhawanInfraStatus: { type: S, allowNull: true },
  gpBhawanEnergyMeter: { type: S, allowNull: true },
  gpBhawanEarthing:    { type: S, allowNull: true },
  gpBhawanSolar:       { type: S, allowNull: true },
  gpBhawanLat:         { type: F, allowNull: true },
  gpBhawanLong:        { type: F, allowNull: true },

  // Section D — Proposed Location
  proposedBuilding:    { type: S, allowNull: true },
  proposedRackSpace:   { type: S, allowNull: true },
  proposedLat:         { type: F, allowNull: true },
  proposedLong:        { type: F, allowNull: true },
  proposedEnergyMeter: { type: S, allowNull: true },
  proposedEarthing:    { type: S, allowNull: true },
  proposedSolar:       { type: S, allowNull: true },
  proposedPoleLength:  { type: S, allowNull: true },
  proposedPoleLat:     { type: F, allowNull: true },
  proposedPoleLong:    { type: F, allowNull: true },
  proposedRemarks:     { type: DataTypes.TEXT, allowNull: true },

  // Sarpanch
  sarpanchName:    { type: S, allowNull: true },
  sarpanchContact: { type: S, allowNull: true },

  // Photo
  photoBase64:     { type: DataTypes.TEXT('long'), allowNull: true },

  userId:   { type: DataTypes.INTEGER, allowNull: true },
  syncedAt: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
}, {
  timestamps: true,
});

module.exports = Survey;

File: spring-backend/src/main/java/com/gpsurvey/backend/models/Survey.java
package com.gpsurvey.backend.models;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
@Entity
@Data
public class Survey {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique=true, nullable=false)
    private String uuid;
    
    @ManyToOne
    @JoinColumn(name="company_id", nullable=false)
    private Company company;

    private Long stateId;
    private String stateName;
    private Long districtId;
    private String districtName;
    private Long blockId;
    private String blockName;
    private Long gramPanchayatId;
    private String gramPanchayatName;
    private String gramPanchayatCode;

    private String phase;
    private String surveyVendor;
    private String surveyDate;
    private String surveyDone = "YES";
    @Column(columnDefinition="TEXT")
    private String remarks;

    private String origLocationType;
    private String origInfraStatus;
    private String origElectricity;
    private String origPowerHours;
    private String origSolar;
    private String origEarthing;
    private Double origLat;
    private Double origLong;

    private String currentLocation;
    private String currentPermTemp;
    private Double currentLat;
    private Double currentLong;

    private String gpBhawanAvailable;
    private String gpBhawanInfraStatus;
    private String gpBhawanEnergyMeter;
    private String gpBhawanEarthing;
    private String gpBhawanSolar;
    private Double gpBhawanLat;
    private Double gpBhawanLong;

    private String proposedBuilding;
    private String proposedRackSpace;
    private Double proposedLat;
    private Double proposedLong;
    private String proposedEnergyMeter;
    private String proposedEarthing;
    private String proposedSolar;
    private String proposedPoleLength;
    private Double proposedPoleLat;
    private Double proposedPoleLong;
    @Column(columnDefinition="TEXT")
    private String proposedRemarks;

    private String sarpanchName;
    private String sarpanchContact;

    @Column(columnDefinition="LONGTEXT")
    private String photoBase64;

    @ManyToOne
    @JoinColumn(name="user_id", nullable=false)
    private User user;
    
    private LocalDateTime syncedAt = LocalDateTime.now();
    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
}
